### DISCUSSION TITLE
Annotations for line and area charts

### DISCUSSION THREAD

**Discussion by priard** [tremorlabs/tremor-npm#591] [Ideas]
[Apex offers an interesting solution for adding annotations](https://apexcharts.com/react-chart-demos/line-charts/line-chart-annotations/) to various parts of the data. One thing I find lacking is a way to indicate, for example, on which day a change occurred or when there was a significant change worth noting in the chart.

![image](https://github.com/tremorlabs/tremor/assets/8668816/915027d1-1c6a-4e52-b50d-f9e56e7c854c)

You might want to consider incorporating this functionality, perhaps in a simplified form, into line and area charts. For instance, annotations could be made with vertical or horizontal lines that can be set at any point on the chart.

**Here are 3 types of annotations to consider in a simple form:**
1. A vertical line set for the selected value, complete with a label.
2. A horizontal line, labeled for the chosen value.
3. Point annotation for the selected value from the x and y axes (also labeled).

There's also the possibility of setting more than one annotation on the chart, and different types simultaneously.

**For individual annotations, you should be able to configure:**
- Label text;
- Color of the line (or points) and label (colors defined by Tremor);
- Ability to hide the label (true/false);
- For point annotations, consider the possibility to show an optional icon.

Let me know what you think about this suggestion. Just tossing an idea into the mix! ;)

**Comment by lacymorrow** [comment]
This could cover the feature requested here: https://github.com/tremorlabs/tremor/issues/443
